packages/nuxt/src/components/runtime/server-component.ts

```diff
@@ -7,10 +7,12 @@ export const createServerComponent = (name: string) => {
     inheritAttrs: false,
     props: { lazy: Boolean },
     setup (props, { attrs, slots }) {
+      // #23051 - remove data-v attributes
+      const attrsWithoutVueDataAttr = computed(() => Object.entries(attrs).reduce<Record<string, string>>((acc, [key, value]) => key.startsWith('data-v-') ? acc : Object.assign(acc, { [key]: value }), {}))
```

Comment:
I think we should do this within the render function, and I think it would likely be simpler to do `Object.fromEntries` + `Object.entries().filter`.